src/transformers/models/llama/modeling_llama.py

```diff
@@ -987,7 +987,9 @@ def forward(
         if position_ids is None:
             position_ids = cache_position.unsqueeze(0)
 
-        causal_mask = self._update_causal_mask(attention_mask, inputs_embeds, cache_position)
+        causal_mask = self._update_causal_mask(
+            attention_mask, inputs_embeds, cache_position, past_seen_tokens + inputs_embeds.shape[1]
```

Comment:
I guess passing `cache_position[-1]` doesn't work? The only issue is that `past_seen_tokens` s gonna be deprecated in favor of cache positions, but that works since with static cache we use the max positions embeddings